staker/bold_state_provider.go

```diff
@@ -0,0 +1,498 @@
+// Copyright 2023, Offchain Labs, Inc.
+// For license information, see https://github.com/offchainlabs/bold/blob/main/LICENSE
+package staker
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/metrics"
+
+	protocol "github.com/OffchainLabs/bold/chain-abstraction"
+	"github.com/OffchainLabs/bold/containers/option"
+	l2stateprovider "github.com/OffchainLabs/bold/layer2-state-provider"
+	"github.com/OffchainLabs/bold/state-commitments/history"
+
+	"github.com/offchainlabs/nitro/arbutil"
+	challengecache "github.com/offchainlabs/nitro/staker/challenge-cache"
+	"github.com/offchainlabs/nitro/validator"
+)
+
+var (
+	_ l2stateprovider.ProofCollector          = (*BOLDStateProvider)(nil)
+	_ l2stateprovider.L2MessageStateCollector = (*BOLDStateProvider)(nil)
+	_ l2stateprovider.MachineHashCollector    = (*BOLDStateProvider)(nil)
+	_ l2stateprovider.ExecutionProvider       = (*BOLDStateProvider)(nil)
+)
+
+var executionNodeOfflineGauge = metrics.NewRegisteredGauge("arb/state_provider/execution_node_offline", nil)
+
+var (
+	ErrChainCatchingUp = errors.New("chain catching up")
+)
+
+type BOLDStateProvider struct {
```

Comment:
have a separate config struct for validatorName, checkBatchfinality, challengeLeafHeights - possibly split that last one in a way more useful for challenge mannager as well (block-challenge/big-step/small-step)
There is certainly no need for the opts in NewBOLDState just to set checkBatchFinality to false

Reply:
Done.